src/serac/physics/solid.cpp

```diff
@@ -397,13 +398,69 @@ void Solid::advanceTimestep(double& dt)
   mesh_.NewNodes(*deformed_nodes_);
 
   cycle_ += 1;
+
+  previous_solve_ = PreviousSolve::Forward;
+}
+
+mfem::ParLinearForm& Solid::shearModulusSensitivity(mfem::ParFiniteElementSpace& shear_space)
+{
+  SLIC_ERROR_ROOT_IF(previous_solve_ == PreviousSolve::None,
+                     "Sensitivities only valid following a forward and adjoint solve.");
+  SLIC_WARNING_ROOT_IF(
+      previous_solve_ == PreviousSolve::Forward,
+      "Sensitivities only valid following a forward and adjoint solve (in that order). The previous solve was a "
+      "forward analysis. Ensure that the correct displacement and adjoint states are set for sensitivies.");
+
+  if (!shear_sensitivity_coef_) {
+    LinearElasticMaterial* linear_mat = dynamic_cast<LinearElasticMaterial*>(material_.get());
+
+    SLIC_ERROR_ROOT_IF(!linear_mat, "Only linear elastic materials allowed for sensitivity analysis.");
```

Comment:
I'm assuming this is temporary, pending further development of `Functional`?

Reply:
Yeah, this is me being lazy. I could do these computations for neo-Hookean as well, but it will become trivially easy once we move the solid module to `Functional`, so I thought my time is better spent elsewhere.